File: backend/app/models/templates.py
from datetime import datetime
from sqlalchemy import Column, Integer, String, Text, Boolean, DateTime, JSON
from app.core.database import Base


class StrategyTemplate(Base):
    __tablename__ = "strategy_templates"

    id               = Column(Integer, primary_key=True, autoincrement=True)
    name             = Column(String(200), nullable=False)
    description      = Column(Text, nullable=True)
    category         = Column(String(100), nullable=True)
    difficulty       = Column(String(20), nullable=True)   # "초급" | "중급" | "고급"
    rule_json        = Column(JSON, nullable=True)          # TradingRule 포맷
    backtest_summary = Column(JSON, nullable=True)          # { cagr, mdd, sharpe }
    tags             = Column(JSON, nullable=True)          # list[str]
    is_active        = Column(Boolean, default=True, nullable=False)
    created_at       = Column(DateTime, default=datetime.utcnow, nullable=False)

File: frontend/src/services/templates.ts
/**
 * 전략 템플릿 서비스 — TODO stub
 * 레거시 백엔드(:8000) 제거. 클라우드 서버 마이그레이션 후 재구현 예정.
 */

export interface BacktestSummary {
  cagr: number
  mdd: number
  sharpe: number
}

export interface TemplateCondition {
  variable: string
  operator: string
  value: number
}

export interface StrategyTemplate {
  id: number
  name: string
  description: string | null
  category: string | null
  difficulty: string | null
  rule_json: { side: string; conditions: TemplateCondition[] } | null
  backtest_summary: BacktestSummary | null
  tags: string[]
}

const STUB_WARN = (name: string) => console.warn(`[stub] ${name}: 레거시 백엔드 제거됨`)

export const templatesApi = {
  list: async (): Promise<StrategyTemplate[]> => { STUB_WARN('templatesApi.list'); return [] },
  get: async (_id: number): Promise<StrategyTemplate> => { STUB_WARN('templatesApi.get'); return {} as StrategyTemplate },
}
